code/src/reducers/game.js

```diff
@@ -0,0 +1,64 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+
+const initialState = {
+    game: {}, 
+    history: [],
+    username: "",
+
+}
+
+export const game = createSlice ({
+    name: 'game', initialState,
+    
+    reducers: {
+    saveGame: (state, action) => {
+        state.history = [...state.history, state.game]
+        state.game = action.payload
+    },
+
+    addUserName: (state, action) => {
+        state.username = action.payload
+    },
+
+    nextStep: (state, action) => {
+        state.username = action.payload
+    }
+}
+})
+
+export const createPlayer = (username) => {
+    return (dispatch) => {
+        fetch('https://wk16-backend.herokuapp.com/start', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({ "username": `${username}` })
+        })
+        .then((res) => res.json())
+        .then((json) => {
+            dispatch(game.actions.saveGame(json))
+        })
+    }
+}
+
+export const continueGame = (step, username) => {
+    return (dispatch) => {
+                fetch('https://wk16-backend.herokuapp.com/action', {
```

Comment:
This shouldn't be as much indented

Reply:
@HIPPIEKICK Have now changed the indentation.